addons/finalAlert/finalAlert.lua

```diff
@@ -0,0 +1,303 @@
+            target = t.id
+        elseif bt then
+            target = bt.id
+        else
+            return
+        end
+
+        if act.category == weapon_skill_category and act.actor_id == target then
+            local skill_name =
+                res.monster_abilities[act.targets[1].actions[1].param] and
+                res.monster_abilities[act.targets[1].actions[1].param].name or
+                "???"
+
+            if act.param == interrupt_id then
+                skill_name = "Interrupted!"
+                show_caption(skill_name, "int")
+            else
+                show_caption(skill_name, "ws")
+            end
+        elseif act.category == magic_category and act.actor_id == target then
+            local spell_name =
+                res.spells[act.targets[1].actions[1].param] and res.spells[act.targets[1].actions[1].param].name or
+                "???"
+
+            if act.param == interrupt_id then
+                spell_name = "Interrupted!"
+                show_caption(spell_name, "int")
+            else
+                show_caption(spell_name, "ma")
+            end
+        end
+    end
+)
+
+function refresh_backgrounds()
+    create_backgrounds(settings.x_position - 250, settings.y_position)
+end
+
+function create_backgrounds(x, y)
+    windower.prim.create(background_ability)
+    windower.prim.set_fit_to_texture(background_ability, true)
+    windower.prim.set_texture(
+        background_ability,
+        windower.addon_path .. "images/" .. settings.background_size .. "/background_ability.png"
+    )
+    windower.prim.set_position(background_ability, x, y)
+    windower.prim.set_visibility(background_ability, false)
+
+    windower.prim.create(background_magic)
+    windower.prim.set_fit_to_texture(background_magic, true)
+    windower.prim.set_texture(
+        background_magic,
+        windower.addon_path .. "images/" .. settings.background_size .. "/background_magic.png"
+    )
+    windower.prim.set_position(background_magic, x, y)
+    windower.prim.set_visibility(background_magic, false)
+
+    windower.prim.create(background_interrupt)
+    windower.prim.set_fit_to_texture(background_interrupt, true)
+    windower.prim.set_texture(
+        background_interrupt,
+        windower.addon_path .. "images/" .. settings.background_size .. "/background_interrupt.png"
+    )
+    windower.prim.set_position(background_interrupt, x, y)
+    windower.prim.set_visibility(background_interrupt, false)
+
+    windower.prim.create(background_emphasize)
+    windower.prim.set_fit_to_texture(background_emphasize, true)
+    windower.prim.set_texture(
+        background_emphasize,
+        windower.addon_path .. "images/" .. settings.background_size .. "/background_emphasize.png"
+    )
+    windower.prim.set_position(background_emphasize, x, y)
+    windower.prim.set_visibility(background_emphasize, false)
+end
+
+function show_caption(text, type)
+    local event_type
+
+    hide_caption()
```

Comment:
Feels weird that you hide it only to show it again. If this is about hiding the currently active box, I'd either maintain which box is currently active in its own variable, or extract the box-hiding code into a new function and call that from both here and `hide_caption`.